App/src/Service/GroupExpenseBalancer.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\Bilan;
+use App\Entity\Expense;
+
+class GroupExpenseBalancer
+{
+    public function expenseBalancer($expenses)
+    {
+        /**
+         * @var Bilan[]
+         */
+        $bilans = array_reduce($expenses, static function (array $bilans, Expense $expense) {
+            $payer = $expense->getPayer();
+            if (!array_key_exists($payer, $bilans)) {
+                return [...$bilans, $payer => new Bilan($payer)];
+            }
+
+            return $bilans;
+        }, []);
```

Comment:
You can reduce this part with a lambda like :
```php
static fn (array $carry, object $item) => array_key_exists($item->anything(), $carry)
    ? $carry
    : [...$carry, $item->anything() => new Object()]
```

Reply:
Ok, but you will explain this to me :)